### PR TITLE
restructure

### PR DESCRIPTION
**Pull request by amir-arad**
mostly rewrite main execution loop: treat different jobs in boyar as services

### CODE REVIEW

File: services/config_poll.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	"github.com/orbs-network/boyarin/supervized"
+	"github.com/orbs-network/boyarin/utils"
+	"github.com/orbs-network/scribe/log"
+	"time"
+)
+
+type ConfigurationPollService struct {
+	flags       *config.Flags
+	logger      log.Logger
+	output      chan config.NodeConfiguration
```
Comment (netoneko) [inline services/config_poll.go]: This seems confusing

File: supervized/supervize_test.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ func TestGoForever(t *testing.T) {
 	maxIterations := 5
 	numOfIterationsBeforeCrash := 2
 
-	GoForever(func() {
+	GoForever(func(_ bool) {
```
Comment (netoneko) [inline supervized/supervize_test.go]: It would be easier to just move to govnr at this point.

File: services/report_status.go [hunk 3]
```diff
@@ -1,15 +1,38 @@
...
+	"github.com/orbs-network/boyarin/supervized"
 	"github.com/orbs-network/scribe/log"
 	"time"
 )
 
-func ReportStatus(ctx context.Context, logger log.Logger, since time.Duration) error {
+const SERVICE_STATUS_REPORT_PERIOD = 1 * time.Minute
+const SERVICE_STATUS_REPORT_TIMEOUT = 30 * time.Second
+
+func WatchAndReportServicesStatus(logger log.Logger) {
```
Comment (netoneko) [inline services/report_status.go]: You can pass the list of services and get rid of `GetVcidFromServiceName` altogether.

File: services/config_poll.go (hunk 1 above)
Comment (amir-arad) [reply services/config_poll.go]: could you explain? I know it's subjective, but what's the source of confusion?

Comment (netoneko) [reply services/config_poll.go]: You have two fields called `output` and `Output`. I can't really understand why there are two channels with the same name.

Comment (amir-arad) [reply services/config_poll.go]: both fields are the same channel (at least I hope so).
One field is private (`output`) for writes. the second field is public (`Channel`) and is read-only.

**Pull request merged**